File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        unordered_set<int> set;
        for (int num : nums){
            set.insert(num);
        }
        for (int i = 0;i<=nums.size();i++){
            if (!set.contains(i)){
                return i;
            } 
        }
        return -1;
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        Set<Integer> numberSet = new HashSet<>();
        for (int i : nums){
            numberSet.add(i);
        }
        for (int i = 0; i<=nums.length;i++){
            if (!numberSet.contains(i)){
                return i;
            }
        }
        return -1;
        
    }
}